discovery-handlers/onvif/src/credential_store.rs

```diff
@@ -0,0 +1,718 @@
+use akri_discovery_utils::discovery::v0::ByteData;
+use std::collections::HashMap;
+
+/// Key name of device credential list in discoveryProperties
+pub const DEVICE_CREDENTIAL_LIST: &str = "device_credential_list";
+/// Key name of device credential ref list in discoveryProperties
+pub const DEVICE_CREDENTIAL_REF_LIST: &str = "device_credential_ref_list";
+/// Key name prfix of username credential list in discoveryProperties
+pub const DEVICE_CREDENTIAL_USERNAME_PREFIX: &str = "username_";
+/// Key name prfix of password credential list in discoveryProperties
+pub const DEVICE_CREDENTIAL_PASSWORD_PREFIX: &str = "password_";
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct CredentialData {
+    username: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    password: Option<String>,
+    #[serde(default = "default_base64encoded")]
```

Comment:
the default of `bool` type is already `false`, no need to override it

Reply:
remove default setter